stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MediaAttachmentView.kt

```diff
@@ -73,14 +82,36 @@ internal class MediaAttachmentView : ConstraintLayout {
         }
 
         if (attachment.type == ModelType.attach_giphy) {
-            binding.imageView.updateLayoutParams {
-                height = style.giphyHeight
+            attachment.giphyInfo(GiphyInfoType.ORIGINAL)?.let { giphyInfo ->
+                containerView?.doOnPreDraw { container ->
+                    val replyView = containerView.findViewById<MessageReplyView>(R.id.replyView)
+
+                    container.updateLayoutParams {
+                        height = parseHeight(giphyInfo) + parseReplyHeight(replyView)
+                        width = parseWidth(giphyInfo)
+                    }
+
+                    binding.imageView.updateLayoutParams {
+                        height = parseHeight(giphyInfo) - DEFAULT_MARGIN_FOR_CONTAINER_DP.dpToPx()
+                        width = parseWidth(giphyInfo) - DEFAULT_MARGIN_FOR_CONTAINER_DP.dpToPx()
+                    }
+
+                    showImageByUrl(url) {
+                        showMore()
+                        showGiphyLabel()
```

Comment:
I see three doublings. Can we extract it to something like `showImageByUrlDefault(url: String)`?